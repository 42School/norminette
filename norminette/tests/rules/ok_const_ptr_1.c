int	func(void)
{
	int const							*var = 0;
	const int							*var = 0;
	t_obj const							*var = 0;
	int *const *const *const *const		var = 0;
	int **const							var = 0;
	int const *const **const			var = 0;
	const t_obj							*var = 0;
	t_obj *const *const *const *const	var = 0;
	t_obj **const						var = 0;
	t_obj const *const **const			var = 0;
}
