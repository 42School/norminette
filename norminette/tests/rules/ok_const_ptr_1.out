[36mok_const_ptr_1.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 1":
		<INT> <TAB> <IDENTIFIER=func> <LPARENTHESIS> <VOID> <RPARENTHESIS> <NEWLINE>
[36mok_const_ptr_1.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 2":
		<LBRACE> <NEWLINE>
[36mok_const_ptr_1.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 3":
		<TAB> <INT> <SPACE> <CONST> <TAB> <TAB> <TAB> <TAB> <TAB> <TAB> <TAB> <MULT> <IDENTIFIER=var> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_1.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 4":
		<TAB> <CONST> <SPACE> <INT> <TAB> <TAB> <TAB> <TAB> <TAB> <TAB> <TAB> <MULT> <IDENTIFIER=var> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_1.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 5":
		<TAB> <IDENTIFIER=t_obj> <SPACE> <CONST> <TAB> <TAB> <TAB> <TAB> <TAB> <TAB> <TAB> <MULT> <IDENTIFIER=var> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_1.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 6":
		<TAB> <INT> <SPACE> <MULT> <CONST> <SPACE> <MULT> <CONST> <SPACE> <MULT> <CONST> <SPACE> <MULT> <CONST> <TAB> <TAB> <IDENTIFIER=var> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_1.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 7":
		<TAB> <INT> <SPACE> <MULT> <MULT> <CONST> <TAB> <TAB> <TAB> <TAB> <TAB> <TAB> <TAB> <IDENTIFIER=var> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_1.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 8":
		<TAB> <INT> <SPACE> <CONST> <SPACE> <MULT> <CONST> <SPACE> <MULT> <MULT> <CONST> <TAB> <TAB> <TAB> <IDENTIFIER=var> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_1.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 9":
		<TAB> <CONST> <SPACE> <IDENTIFIER=t_obj> <TAB> <TAB> <TAB> <TAB> <TAB> <TAB> <TAB> <MULT> <IDENTIFIER=var> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_1.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 10":
		<TAB> <IDENTIFIER=t_obj> <SPACE> <MULT> <CONST> <SPACE> <MULT> <CONST> <SPACE> <MULT> <CONST> <SPACE> <MULT> <CONST> <TAB> <IDENTIFIER=var> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_1.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 11":
		<TAB> <IDENTIFIER=t_obj> <SPACE> <MULT> <MULT> <CONST> <TAB> <TAB> <TAB> <TAB> <TAB> <TAB> <IDENTIFIER=var> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_1.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 12":
		<TAB> <IDENTIFIER=t_obj> <SPACE> <CONST> <SPACE> <MULT> <CONST> <SPACE> <MULT> <MULT> <CONST> <TAB> <TAB> <TAB> <IDENTIFIER=var> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_1.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 13":
		<RBRACE> <NEWLINE>
ok_const_ptr_1.c: Error!
Error: TOO_MANY_VARS_FUNC   (line:   8, col:   1):	Too many variables declarations in a function
Error: TOO_MANY_VARS_FUNC   (line:   9, col:   1):	Too many variables declarations in a function
Error: TOO_MANY_VARS_FUNC   (line:  10, col:   1):	Too many variables declarations in a function
Error: TOO_MANY_VARS_FUNC   (line:  11, col:   1):	Too many variables declarations in a function
Error: TOO_MANY_VARS_FUNC   (line:  12, col:   1):	Too many variables declarations in a function
